src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -404,6 +409,8 @@ class ToolLoader:
     def __init__(self, working_dir: str, package_tool_keys: Optional[List[str]] = None) -> None:
         self._working_dir = working_dir
         self._package_tools = collect_package_tools(package_tool_keys) if package_tool_keys else {}
+        # Used to handle backward compatibility of tool ID changes.
+        self._tool_id_mapping = build_tool_id_mapping(self._package_tools)
```

Comment:
Maybe `_transitioning_tools`? which is more straightforward with "transition_from"

Reply:
Or `_deprecated_tools`

Reply:
Changed.